codestodians/food-basket-front-end
# [Feature] Access history from component props

#### What does this PR do?
- Ensures that the history object is accessed from the component props

#### Description of Task to be completed?
- Switch to the use of `BrowserRouter`.
- Access pathname from the `location` prop in any component that needs it.
- Create helpers to aid in testing.
- Fix failing tests.

#### How should this be manually tested?
- Check out into the `feature/access-history-from-component-props-158144890` branch.
- Run the app, as before, it should work perfectly.

#### Extra Info (Test coverage)?
- The test coverage drops as a result of this story; some lines will remain untested, and this is because it is very difficult to sufficiently test the browser location in a node environment. After persisting and making enquiries from a couple of persons, it was suggested to me that most people do not expend extra effort trying to test the lines that involve location and history as a result of the aforementioned difficulty.

#### What are the relevant pivotal tracker stories?
[#158144890](https://www.pivotaltracker.com/story/show/158144890)
